File: app/src/main/java/com/example/inappcamerademo/activity/ImageDetailedViewActivity.java
package com.example.inappcamerademo.activity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.inappcamerademo.MainDbHelper;
import com.example.inappcamerademo.R;
import com.example.inappcamerademo.Utils;
import com.example.inappcamerademo.model.ImageInfo;
import com.google.android.gms.maps.CameraUpdateFactory;
import com.google.android.gms.maps.GoogleMap;
import com.google.android.gms.maps.OnMapReadyCallback;
import com.google.android.gms.maps.SupportMapFragment;
import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.MarkerOptions;

import java.io.File;
import java.io.FileInputStream;

public class ImageDetailedViewActivity extends AppCompatActivity implements OnMapReadyCallback {

    private ImageInfo imageInfo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_image_details);

        SupportMapFragment mapFragment = (SupportMapFragment) getSupportFragmentManager().findFragmentById(R.id.image_map_location);
        mapFragment.getMapAsync(this);

        getSupportActionBar().setHomeButtonEnabled(true);
        getSupportActionBar().setTitle("Information");

        Intent intent = getIntent();

        String fileName = intent.getStringExtra("name");
        MainDbHelper mainDbHelper = MainDbHelper.getInstance(this.getApplicationContext());

        imageInfo = mainDbHelper.getImageInfoFromImageName(fileName);

        File imageFile = new File(imageInfo.getPath());
        Bitmap bitmap;

        try {
            bitmap = BitmapFactory.decodeStream(new FileInputStream(imageFile));
            ((ImageView) findViewById(R.id.captured_image)).setImageBitmap(bitmap);
        } catch (Exception e) {
            e.printStackTrace();
        }

        ((TextView) findViewById(R.id.image_name)).setText(String.format("File name: %s", imageInfo.getName()));
        ((TextView) findViewById(R.id.image_time)).setText(String.format("Time: %s", Utils.df.format(imageInfo.getTimestamp() * 1000)));
        ((TextView) findViewById(R.id.image_path)).setText(String.format("File path: %s", imageInfo.getPath()));
    }


    @Override
    public void onMapReady(GoogleMap googleMap) {

        // Add a marker in Sydney and move the camera
        if (imageInfo.getLat() != 0 && imageInfo.getLat() != 0) {
            LatLng latLng = new LatLng(imageInfo.getLat(), imageInfo.getLon());
            googleMap.addMarker(new MarkerOptions().position(latLng));
            googleMap.moveCamera(CameraUpdateFactory.newLatLng(latLng));
            googleMap.setMinZoomPreference(15);
        }
    }
}

File: app/src/main/java/com/example/inappcamerademo/activity/ImageDetailedViewActivity.kt
package com.example.inappcamerademo.activity

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.inappcamerademo.MainDbHelper
import com.example.inappcamerademo.R
import com.example.inappcamerademo.model.ImageInfo
import com.example.inappcamerademo.utils.Utils
import com.google.android.gms.maps.CameraUpdateFactory
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.OnMapReadyCallback
import com.google.android.gms.maps.SupportMapFragment
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions
import dagger.hilt.android.AndroidEntryPoint
import java.io.File
import java.io.FileInputStream

class ImageDetailedViewActivity : AppCompatActivity(), OnMapReadyCallback {
    private var imageInfo: ImageInfo? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_image_details)

        val mapFragment =
            supportFragmentManager.findFragmentById(R.id.image_map_location) as SupportMapFragment?

        mapFragment?.let {
            it.getMapAsync(this)
        }

        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setTitle("Information")

        val mainDbHelper = MainDbHelper.getInstance(this.applicationContext)

        intent.getStringExtra("name")?.also {
            imageInfo = mainDbHelper?.getImageInfoFromImageName(it)
        }

        imageInfo?.path?.let {
            val imageFile = File(it)
            val bitmap: Bitmap
            try {
                bitmap = BitmapFactory.decodeStream(FileInputStream(imageFile))
                (findViewById<View>(R.id.captured_image) as ImageView).setImageBitmap(bitmap)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }
        (findViewById<View>(R.id.image_name) as TextView).text =
            String.format("File name: %s", imageInfo?.name)
        (findViewById<View>(R.id.image_time) as TextView).text =
            String.format("Time: %s", Utils.df.format(imageInfo?.timestamp?.times(1000)))
        (findViewById<View>(R.id.image_path) as TextView).text =
            String.format("File path: %s", imageInfo?.path)
    }

    override fun onMapReady(googleMap: GoogleMap) {

        // Add a marker in Sydney and move the camera
        if (imageInfo!!.lat != 0.0 && imageInfo!!.lat != 0.0) {
            val latLng = LatLng(imageInfo!!.lat, imageInfo!!.lon)
            googleMap.addMarker(MarkerOptions().position(latLng))
            googleMap.moveCamera(CameraUpdateFactory.newLatLng(latLng))
            googleMap.setMinZoomPreference(15f)
        }
    }

}